units/lru_container_removal.go

```diff
@@ -0,0 +1,114 @@
+package units
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/cloud"
+	"github.com/evergreen-ci/evergreen/model/host"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/pkg/errors"
+)
+
+const (
+	leastRecentlyUsedImageJobName = "least-recently-used-image"
+	// if each image is around 11GB, allow 10 images at a time
+	maxDiskUsage = 1024 * 1024 * 1024 * 11 * 10
+)
+
+func init() {
+	registry.AddJobType(leastRecentlyUsedImageJobName, func() amboy.Job {
+		return makeLeastRecentlyUsedImageTimeJob()
+	})
+
+}
+
+type leastRecentlyUsedImageJob struct {
+	HostID   string `bson:"host_id" json:"host_id" yaml:"host_id"`
+	job.Base `bson:"base" json:"base" yaml:"base"`
+
+	// cache
+	host     *host.Host
+	env      evergreen.Environment
+	provider string
+	settings *evergreen.Settings
+}
+
+func makeLeastRecentlyUsedImageTimeJob() *leastRecentlyUsedImageJob {
+	j := &leastRecentlyUsedImageJob{
+		Base: job.Base{
+			JobType: amboy.JobType{
+				Name:    leastRecentlyUsedImageJobName,
+				Version: 0,
+			},
+		},
+	}
+	j.SetDependency(dependency.NewAlways())
+
+	return j
+}
+
+func NewLeastRecentlyUsedImageJob(h *host.Host, providerName, id string) amboy.Job {
+	j := makeLeastRecentlyUsedImageTimeJob()
+
+	j.host = h
+	j.provider = providerName
+	j.HostID = h.Id
+
+	j.SetID(fmt.Sprintf("%s.%s", leastRecentlyUsedImageJobName, id))
+	return j
+}
+
+func (j *leastRecentlyUsedImageJob) Run(ctx context.Context) {
+	var cancel context.CancelFunc
+
+	ctx, cancel = context.WithCancel(ctx)
```

Comment:
There's no need to add a canceler here unless you're going to explicitly cancel the context at some point